tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ClientCredentialsTests.cs

```diff
@@ -178,6 +179,44 @@ public async Task WithOnBeforeTokenRequest_TestAsync(Cloud cloud, TargetFramewor
             Assert.AreEqual(TokenSource.Cache, authResult.AuthenticationResultMetadata.TokenSource);            
         }
 
+        [TestMethod]
+        public async Task ByRefreshTokenTestAsync()
+        {
+            // Arrange
+            var labResponse = await LabUserHelper.GetDefaultUserAsync().ConfigureAwait(false);
+
+            var msalPublicClient = PublicClientApplicationBuilder
+                .Create(labResponse.App.AppId)
+                .WithTestLogging()
+                .WithAuthority(labResponse.Lab.Authority, "organizations")
+                .BuildConcrete();
+
+            AuthenticationResult authResult = await msalPublicClient
+                .AcquireTokenByUsernamePassword(s_scopes, labResponse.User.Upn, labResponse.User.GetOrFetchPassword())
+                .ExecuteAsync(CancellationToken.None)
+                .ConfigureAwait(false);
+
+            var confidentialApp = ConfidentialClientApplicationBuilder
```

Comment:
Hmm, you can build CCA without secret or cert?

Reply:
Dont think so, i think I did control z one too many times when cleaning up the code. Ill add back the credential.